### PR TITLE
uhdm-plugin: add support for wires with custom type and multiple ranges

### PR DESCRIPTION
**Pull request by kamilrakoczy**
This PR adapts enums in structs to have same syntax as mainline yosys and adds support for wires with custom type and multiple ranges.

### CODE REVIEW

File: uhdm-plugin/UhdmAst.cc [hunk 1]
```diff
@@ -381,6 +394,46 @@ AST::AstNode *UhdmAst::make_ast_node(AST::AstNodeType type, std::vector<AST::Ast
...
+    visit_one_to_one({vpiElemTypespec}, obj_h, [&](AST::AstNode *node) {
+        if (node && node->type == AST::AST_STRUCT) {
+            auto str = current_node->str;
+            node->cloneInto(current_node);
+            current_node->str = str;
+            delete node;
+        } else if (node) {
+            current_node->str = node->str;
+#ifdef BUILD_UPSTREAM
+            if (node->type == AST::AST_ENUM) {
```
Comment (kbieganski) [inline uhdm-plugin/UhdmAst.cc:413]: Should be one `if`.

File: uhdm-plugin/UhdmAst.cc [hunk 2]
```diff
@@ -1269,25 +1333,85 @@ AST::AstNode *UhdmAst::convert_dot(AST::AstNode *node, AST::AstNode *dot, AST::A
     log_assert(shared.current_top_node != nullptr);
     AST_INTERNAL::current_ast_mod = shared.current_top_node;
     visitEachDescendant(shared.current_top_node, [&](AST::AstNode *current_scope_node) {
-        if (current_scope_node->type == AST::AST_TYPEDEF || current_scope_node->type == AST::AST_WIRE) {
+        if (current_scope_node->type == AST::AST_TYPEDEF || current_scope_node->type == AST::AST_WIRE || current_scope_node->type == AST::AST_ENUM) {
             AST_INTERNAL::current_scope[current_scope_node->str] = current_scope_node;
         }
     });
+    if (!wire_node->children.empty()) {
```
Comment (kbieganski) [inline uhdm-plugin/UhdmAst.cc:1340]: Should be one `if`.

File: uhdm-plugin/UhdmAst.cc (hunk 1 above)
Comment (kamilrakoczy) [reply uhdm-plugin/UhdmAst.cc]: Fixed.

File: uhdm-plugin/UhdmAst.cc (hunk 2 above)
Comment (kamilrakoczy) [reply uhdm-plugin/UhdmAst.cc]: Fixed.

**Review by kbieganski — APPROVED** [review]

**Pull request merged**